zero_sum_eval/caching.py

```diff
@@ -0,0 +1,27 @@
+from typing import Optional
+from pathlib import Path
+
+import os
+import dspy
+
+_DEFAULT_CACHE_DIR = os.path.join(Path().home(), ".zse_cache")
+
+def cache_module(module: dspy.Module, model: str, role: str, optimizer: str, dataset: str, cache_dir: Optional[str] = None) -> None:
```

Comment:
can we call this save_module instead?

saving is more general than caching, as caching would be a special case.